### ISSUE TITLE
[Discussion] Alternate PDF reader engine (like Sioyek)

### ISSUE THREAD

**Issue by ryanwwest** [elias-sundqvist/obsidian-annotator#282]
Academics and vim enthusiasts such as myself enjoy using https://github.com/ahrm/sioyek, a lightweight PDF reader with some ingenious, keyboard-focused features to make reading/analyzing textbooks and research papers much easier. Sioyek is written in C and itself uses [mupdf](github.com/ArtifexSoftware/mupdf) as its base (also in C), whereas obsidian-annotator uses pdf.js.

Obviously Mupdf and Sioyek, being in C, aren't going to be plug-and-play with this plugin (I presume they'd need some sort of port to Javascript first and work in a browser, which [maybe Mupdf already has](https://github.com/andytango/mupdf-js)). But I think it's great software and hope that someone (perhaps me, someday...) will make it work in a browser and allow its features to be easily used everywhere.

My question is then, assuming someone wants to switch out pdf.js for something else, **what would it take to support a different reader/engine in obsdian-annotator?** I feel like this can't be too difficult since the separate ePUB reader and web/HTML (it's buggy so sorta) already work in this plugin. But maybe I'm way off.

Also, would anyone else also be interested in something like this?

**Comment by astkaasa** [comment]
I want to suggest readium to substitute epub.js again.

**Comment by GlobalCBWK** [comment]
> I want to suggest readium to substitute epub.js again.

Yeah,When I using annotator to read some epubs,Font Size configration totally does nothing
